compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2938,3 +2938,11 @@ class ClosureCannotHaveInternalParameterDependencies(mt: Type)(using Context)
       i"""cannot turn method type $mt into closure
          |because it has internal parameter dependencies"""
     def explain(using Context) = ""
+
+class ImplausiblePatternWarning(pat: tpd.Tree, selType: Type)(using Context)
+  extends TypeMsg(ImplausiblePatternWarningID):
+    def msg(using Context) =
+      i"""|Implausible pattern:
+          |$pat  could match selector of type  $selType
```

Comment:
Not sure if the space is intentional:
```suggestion
          |$pat could match selector of type $selType
```

Reply:
I did that intentionally since I wanted to have some visual space between the program code and the actual text. Let's see how it works in practice.